### PR TITLE
MINOR: remove streams config params from producer/consumer configs

### PR DESCRIPTION
**Pull request by ymatsuda**
Removing streams' specific config params from producer/consumer configs to reduce warning messages.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -273,15 +272,27 @@ public StreamsConfig(Map<?, ?> props) {
         props.put(ProducerConfig.LINGER_MS_CONFIG, "100");
 
         // remove properties that are not required for producers
+        removeStreamsSpecificConfigs(props);
         props.remove(StreamsConfig.KEY_DESERIALIZER_CLASS_CONFIG);
         props.remove(StreamsConfig.VALUE_DESERIALIZER_CLASS_CONFIG);
-        props.remove(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG);
+        props.remove(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG);
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:278]: Is this intentional?

Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:278]: Yes. Any concern?

Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:278]: Do we want to always enforce "latest" for AUTO_OFFSET_RESET_CONFIG? I think there are some cases where a user may want to consume from earliest.

Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:278]: This is a producer config.

Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:278]: Ah, right.

**Comment by guozhangwang** [comment]
LGTM. Filed separate JIRA for transient failure. Merged to trunk.

**Pull request closed without merging**